ersilia/core/tracking.py

```diff
@@ -225,166 +242,170 @@ def upload_to_cddvault(output_df, api_key):
         return False
 
 
-class RunTracker:
+class RunTracker(ErsiliaBase):
+
     """
     This class will be responsible for tracking model runs. It calculates the desired metadata based on a model's
     inputs, outputs, and other run-specific features, before uploading them to AWS to be ingested
     to Ersilia's Splunk dashboard.
     """
-
-    def __init__(self):
+    
+    def __init__(self, model_id, config_json):
+        ErsiliaBase.__init__(self, config_json=config_json, credentials_json=None)
         self.time_start = None
         self.memory_usage_start = 0
+        self.model_id = model_id
+
+        # Initialize folders
+        self.ersilia_runs_folder = os.path.join(EOS, ERSILIA_RUNS_FOLDER)
+        os.makedirs(self.ersilia_runs_folder, exist_ok=True)
+        self.metadata_folder = os.path.join(self.ersilia_runs_folder, "metadata")
+        os.makedirs(self.metadata_folder, exist_ok=True)
+        self.lake_folder = os.path.join(self.ersilia_runs_folder, "lake")
+        os.makedirs(self.lake_folder, exist_ok=True)
+        self.logs_folder = os.path.join(self.ersilia_runs_folder, "logs")
+        os.makedirs(self.logs_folder, exist_ok=True)
+        self.tabular_result_logger = TabularResultLogger()
+        self.PERSISTENT_FILE_PATH = open_persistent_file(model_id)
+        
 
-    # function to be called before model is run
     def start_tracking(self):
         """
-        Runs any code necessary for the beginning of the run.
-        Currently necessary for tracking the runtime and memory usage of a run.
-        """
+	Runs any code necessary for the beginning of the run.
+	Currently necessary for tracking the runtime and memory usage of a run.
+	
+	"""
         self.time_start = datetime.now()
         tracemalloc.start()
         self.memory_usage_start = tracemalloc.get_traced_memory()[0]
-
-    def sample_df(self, df, num_rows, num_cols):
-        """
-        Returns a sample of the dataframe, with the specified number of rows and columns.
+    
+    def calculate_stats(self, result):
```

Comment:
I don't think you should be modifying this method in this PR - it's unrelated and already being worked on by @dzumii